### PR TITLE
Add test helpers for Memcached instance

### PR DESCRIPTION
**Pull request by kajinamit**
This introduces a few helper functions to manage Memcached instance in envtest.

### CODE REVIEW

File: modules/test-operators/helpers/memcached.go
Comment (bshephar) [inline modules/test-operators/helpers/memcached.go:93]: Nice

**Review by bshephar — COMMENTED** [review]
This looks good. It might be nice if we can move Horizon over to using these helper functions and link the PR here for folks to see how these helper functions are implemented.

File: modules/test-operators/helpers/memcached.go
Comment (kajinamit) [inline modules/test-operators/helpers/memcached.go:30]: This is not used. Probably we can just expose MemcachedSpec and use it.

Comment (kajinamit) [reply modules/test-operators/helpers/memcached.go]: Fixed in the latest version.

**Comment by kajinamit** [comment]
> This looks good. It might be nice if we can move Horizon over to using these helper functions and link the PR here for folks to see how these helper functions are implemented.

I can create a PR but the problem is that PR does not pass CI unless we merge this PR. so I'll work on it after we merge this PR unless someone wants to review it as part of reviewing this PR.

File: modules/test-operators/helpers/memcached.go
Comment (gibizer) [reply modules/test-operators/helpers/memcached.go:50]: Probably a remainder of an older pattern where only the namespace was passed in and the name was generated by call itself.

Comment (stuggi) [reply modules/test-operators/helpers/memcached.go:50]: my point was more, we pass in the name and namespace. which is 1:1 set in the namespaced name. so this seem to be not really required to return it. the caller already has all the details. but I am fine with it if it is the pattern we also use in other places

Comment (stuggi) [reply modules/test-operators/helpers/memcached.go:50]: > Probably a remainder of an older pattern where only the namespace was passed in and the name was generated by call itself.

yes probably

**Review by gibizer — APPROVED** [review]
looks good to me

File: modules/test-operators/helpers/memcached.go
Comment (kajinamit) [reply modules/test-operators/helpers/memcached.go:50]: returning the NamespacedName might be redundant, but would allow you to call DeferCleanup directl like `DeferCleanup(th.DeleteInstance, CreateMemcached(...))` or `DeferCleanup(th.DeleteMemcached, CreateMemcached(...))` and we use this pattern quite often. So allowing this interface would be less confusing.

**Pull request merged**
